sdk/core/azure-core/inc/azure/core/context.hpp

```diff
@@ -56,52 +58,46 @@ namespace Azure { namespace Core {
       bool operator!=(Key const& other) const { return !(*this == other); }
     };
 
-    /**
-     * @brief A type used to provide a point in time for context expiration.
-     */
-    using time_point = std::chrono::system_clock::time_point;
-
   private:
     struct ContextSharedState
     {
       std::shared_ptr<ContextSharedState> Parent;
-      std::atomic_int64_t CancelAtMsecSinceEpoch;
+      std::atomic<DateTime::rep> Deadline;
       Context::Key Key;
       std::shared_ptr<void> Value;
       const std::type_info& ValueType;
 
-      static constexpr int64_t ToMsecSinceEpoch(time_point time)
+      static constexpr DateTime::rep ToDateTimeRepresentation(DateTime const& dateTime)
       {
-        return static_cast<int64_t>(
-            std::chrono::duration_cast<std::chrono::milliseconds>(time - time_point()).count());
+        return dateTime.time_since_epoch().count();
       }
 
-      static constexpr time_point FromMsecSinceEpoch(int64_t msec)
+      static constexpr DateTime FromDateTimeRepresentation(DateTime::rep dtRepresentation)
       {
-        return time_point() + static_cast<std::chrono::milliseconds>(msec);
+        return DateTime(DateTime::time_point(DateTime::duration(dtRepresentation)));
       }
 
       explicit ContextSharedState()
-          : CancelAtMsecSinceEpoch(ToMsecSinceEpoch((time_point::max)())), Value(nullptr),
+          : Deadline(ToDateTimeRepresentation((DateTime::max)())), Value(nullptr),
             ValueType(typeid(std::nullptr_t))
       {
       }
 
       explicit ContextSharedState(
           const std::shared_ptr<ContextSharedState>& parent,
-          time_point cancelAt)
-          : Parent(parent), CancelAtMsecSinceEpoch(ToMsecSinceEpoch(cancelAt)), Value(nullptr),
+          DateTime const& deadline)
+          : Parent(parent), Deadline(ToDateTimeRepresentation(deadline)), Value(nullptr),
             ValueType(typeid(std::nullptr_t))
       {
       }
 
       template <class T>
       explicit ContextSharedState(
           const std::shared_ptr<ContextSharedState>& parent,
-          time_point cancelAt,
+          DateTime const& deadline,
```

Comment:
All these changes related to `ContextSharedState` are all private, correct?

Reply:
yes - `ContextSharedState` is `Context`'s private struct.